src/main.js

```diff
@@ -55,23 +61,42 @@ if (crashReporterDirectory) {
 		}
 	}
 
-	// Crashes are stored in the temp directory by default, so we
+	// Crashes are stored in the crashDumps directory by default, so we
 	// need to change that directory to the provided one
-	console.log(`Found --crash-reporter-directory argument. Setting temp directory to be '${crashReporterDirectory}'`);
-	app.setPath('temp', crashReporterDirectory);
-
-	// Start crash reporter
-	const { crashReporter } = require('electron');
-	const productName = (product.crashReporter && product.crashReporter.productName) || product.nameShort;
-	const companyName = (product.crashReporter && product.crashReporter.companyName) || 'Microsoft';
-	crashReporter.start({
-		companyName: companyName,
-		productName: process.env['VSCODE_DEV'] ? `${productName} Dev` : productName,
-		submitURL: '',
-		uploadToServer: false
-	});
+	console.log(`Found --crash-reporter-directory argument. Setting crashDumps directory to be '${crashReporterDirectory}'`);
+	app.setPath('crashDumps', crashReporterDirectory);
+} else {
+	const appCenter = product.appCenter;
+	// If enable-crash-reporter argv is undefined then this is a fresh start,
+	// based on telemetry.enableCrashreporter settings, generate a UUID which
+	// will be used as crash reporter id and also update the json file.
+	if (argvConfig['enable-crash-reporter'] === undefined) {
+		process.argv.push('--create-crash-reporter-config');
+	}
+	// Disable Appcenter crash reporting if
+	// * --crash-reporter-directory is specified
+	// * enable-crash-reporter runtime argument is set to 'false'
+	// * --disable-crash-reporter command line parameter is set
+	if (appCenter && (argvConfig['enable-crash-reporter'] !== false) && !args['disable-crash-reporter']) {
```

Comment:
@deepak1556 isn't this starting the crash reporter even if this is the first startup where `argvConfig['enable-crash-reporter'] === undefined`? Don't we need to wait until we have a crash reporter ID for this to work?

Reply:
oh yeah should fix that, but without crash reporter ID it will be a noop even if called on the server side. But avoiding the call is all the more better.